packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -264,14 +264,27 @@ export class MonacoWorkspace implements lang.Workspace {
     async applyBulkEdit(workspaceEdit: monaco.languages.WorkspaceEdit, options?: EditorOpenerOptions): monaco.Promise<monaco.editor.IBulkEditResult> {
         let totalEdits = 0;
         let totalFiles = 0;
-        const uri2Edits = this.groupEdits(workspaceEdit);
-        for (const uri of uri2Edits.keys()) {
-            const editorWidget = await this.editorManager.open(new URI(uri), options);
+        let versionConflict = false;
+        const doc2Edits = this.groupEdits(workspaceEdit);
+        const editor2Edits = new Map<{ editor: MonacoEditor, model: monaco.editor.ITextModel }, monaco.languages.TextEdit[]>();
+        for (const doc of doc2Edits.keys()) {
+            const editorWidget = await this.editorManager.open(new URI(doc.uri), options);
```

Comment:
I feel like this behaviour is a bit weird, but on the other hand, instead of doing `for ... await` can this be parallelized ?

Although it might be out of the scope of this PR.. Just a concern of mine.

Reply:
That brings up another possible race: We should make sure the versions don't change after checking, so there shouldn't be an await after check and before applying the change. I'll reiterate on that.

Reply:
I have addressed that in the recent force push.